dataflow/flex-templates/streaming_beam_sql/README.md

```diff
@@ -0,0 +1,315 @@
+# Dataflow flex templates - Streaming Beam SQL
+
+[![Open in Cloud Shell](http://gstatic.com/cloudssh/images/open-btn.svg)](https://console.cloud.google.com/cloudshell/editor)
+
+📝 Docs: [Using Flex Templates](https://cloud.google.com/dataflow/docs/guides/templates/using-flex-templates)
+
+Samples showing how to create and run an
+[Apache Beam](https://beam.apache.org/) template with a custom Docker image on
+[Google Cloud Dataflow](https://cloud.google.com/dataflow/docs/).
+
+## Before you begin
```

Comment:
Wow - this is long, I understand your trying to teach here, but I wonder if providing a script that does it and explains as it runs might be easier on users.  `./setup.sh`

Reply:
I agree, it's a lot. I played a bit with creating a script, but now I would have to handle user confirmations when creating/deleting resources and much more validation. That would also make doing the setup/cleanup harder to port to windows. I will leave the instructions in the README for simplicity for now.

These instructions are also present in the docs, so we would prefer to keep them as compatible as possible.

Reply:
Why not link to the docs then?  Though I prefer README's where it's included, but a single source of truth is also a good idea.

Reply:
They want to merge fast for the launch and I don't want to stall the launch since a script like that would take several weeks to develop. I agree that a script would be very handy, but at this point we all just want this to merge and get it over with.

Reply:
Hi @lesv, sorry for my last response, I was a little frustrated with how all the deadlines the product team wanted to meet. I did play around for just under an hour trying to implement a setup script, here's what I learned from it.

* We have to check if a resource already exists before creating it
* We need to ask for user confirmation before creating any resources
* Right now Windows users only have to make small changes to the commands, mainly the `\` for line continuations. But having a bash script would make it much more harder for them to run the sample.
* All the complexity adds up and the script was turning out to be as large as the sample itself
* Also having the explanations in Markdown makes the text clearer to read with all the nice formatting
* Most of the instructions (except for Cloud Scheduler), are actually instructions that users need to follow to use Flex Templates, so it makes sense to have them in the README
* The docs are somewhat simplified to make them simpler and easier to follow, while the README has more detail on some of the commands and why/how to run them, so it makes sense to keep both.

I had a chat with Meredith and we agreed to keep it like this for now. If we see there is a need from users in the future, we can allocate extra resources into creating these scripts.

Reply:
Sg - I already approved the PR